Background worker crash

<!-- Thank you for reporting issues to us! Please try to answer all the questions below. If you don't have an issue to report, you can delete the pre-filled contents and proceed with your comments.  -->

### What are you experiencing an issue with?
<!-- These [ ] are checkboxes, they can be checked like so [x] -->
- [x] Aurora
- [ ] Aurora supported game, (Fill in the game)
- [ ] Aurora Wrapper
- [ ] Aurora Profile
- [ ] Aurora supported device, (Fill in problematic devices)
- [ ] Aurora Scripts
- [ ] Aurora Updater
- [ ] Aurora Lighting issue
- [ ] Other, (Fill in another issue here)

### What kind of an issue is it?
<!-- These [ ] are checkboxes, they can be checked like so [x] -->
- [x] Crash
- [ ] Performance problems
- [ ] Visual bug
- [ ] Localization issue
- [ ] Enhancement suggestion
- [ ] Device issue
- [ ] Other, (Fill in another type of issue here)

### Expected behavior:
Aurora shouldn't crash....?

### Actual behavior:
Upon resuming from sleep

### How often does this problem occur?
Seemingly every time I come back from a long pause

### Please describe the steps to reproduce this issue:
The issue can be reproduced by doing following steps:

1. Run aurora as usual
2. Lock workstation for a short time 
3. Unlock PC

(also happens on resume from sleep or screensaver)

### What is the version of Aurora you are experiencing this issue with?
<!-- Aurora version can be located in settings under the About tab -->
My version is... v0.6.1-dev4

### Does the previous version of Aurora have this issue?
I don't have this issue on my work PC with the same version.

### Please include any logs, screenshots, or files that are related to this issue:
<!-- Log files can be located in the "Aurora Location/logs" directory. -->

Aurora fatally crashed. Please report the follow to author: 
System.InvaIidOperationException: This BackgroundWorker is currently 
busy and cannot run multiple tasks concurrently. 
System. ComponentModeI. BackgroundWorker RunWorkerAsync(Object 
argument) 
at Aurora. Devices. DeviceContainer. < ctor> sender, 
RunWorkerCompIetedEventArgs args) in 
er.cs:line 33 
System. ComponentModeI. BackgroundWorker .0nRunWorkerCompIete 
d(RunWorkerCompIetedEventArgs e) 
at System. Threading. Execution Context. RunlnternaI(ExecutionContext 
executionContext, ContextCaIIback callback* Object state, Boolean 
preserveSyncCtx) 
at System. Threa ding. ExecutionContext. Run (ExecutionContext 
executionContext, ContextCaIIback callback* Object state, Boolean 
preserveSyncCtx) 
System. Threading.QueueLlserWorkItemCaIIback. System. Threading. IThre 
adPooIWorkItem.ExecuteWorkItem0 
at System.Threading.ThreadPooIWorkQueue. Dispatch0

Well, the 'crashes' are definitely rarer now, yesterday I didn't have any and today I only had one...

`[ERROR] 14:42:25: Fatal Exception caught : System.InvalidOperationException: This BackgroundWorker is currently busy and cannot run multiple tasks concurrently.
   at System.ComponentModel.BackgroundWorker.RunWorkerAsync(Object argument)
   at Aurora.Devices.DeviceContainer.<.ctor>b__15_0(Object sender, RunWorkerCompletedEventArgs args) in C:\projects\aurora\Project-Aurora\Project-Aurora\Devices\DeviceManager.cs:line 33
   at System.ComponentModel.BackgroundWorker.OnRunWorkerCompleted(RunWorkerCompletedEventArgs e)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.QueueUserWorkItemCallback.System.Threading.IThreadPoolWorkItem.ExecuteWorkItem()
   at System.Threading.ThreadPoolWorkQueue.Dispatch()
[ERROR] 14:42:25: Runtime terminating: True
[INFO] 15:20:51: Show Window called`

--- a/Project-Aurora/Project-Aurora/Devices/DeviceManager.cs
+++ b/Project-Aurora/Project-Aurora/Devices/DeviceManager.cs
@@ -28,7 +28,7 @@ namespace Aurora.Devices
             Worker.DoWork += WorkerOnDoWork;
             Worker.RunWorkerCompleted += (sender, args) =>
             {
-                if (newFrame)
+                if (newFrame && !Worker.IsBusy)
                     Worker.RunWorkerAsync();
             };
             Worker.WorkerSupportsCancellation = true;

